### PR TITLE
Feature/rmf 81 deploy mch aws

### PR DESCRIPTION
**Pull request by nestabur**
To do still within PR:
- [x] Need to add instructions to develop in CSCS locally, via empty (bare) conda env + poetry on top.
- [x] Remove docs folder and remove TODO re documentation in Jenkinsfile
- [x] No need to use pypi global repo since open source but not open to collaboration
- [x] Before merge need to tag the current main for CSCS operations as-is
- [x] Comment on Jenkinsfile that this is currently only for AWS deployment
- [x] Add back into the readme of this branch - note about 

> uncomment the line etc...

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
...
+
+- Integrate poetry dependency management substituting conda development environments (following MCH python templates)
+- Automate distributable package generation using poetry
+- Configure CI/CD pipeline to run in MCH Jenkins server
+- Containerize pyflexplot tool to be able to run it within a container
+- Integrate AWS deployment
+
 ## 1.0.10 (2023-XX-XX)
 
 - First release on PyPI.
```
Comment (pirmink) [inline HISTORY.md:13]: Has v1.x ever been released on PyPI? Not as far as I now, unless @nestabur made a release of 1.0.10 on PyPI (as the date is not set, probably not?).

File: src/pyflexplot/__init__.py
Comment (pirmink) [inline src/pyflexplot/__init__.py:1]: While Stefan is undisputedly the main author of this code, I wonder if it wouldn't make sense to include a MeteoSwiss contact here?

**Review by pirmink — CHANGES_REQUESTED** [review]
I leave the main functionality to @victoria-cherkas to review. I mainly looked at the README.md file and added some parts that need to be retained from the percursor-versions (README.md and README.meteoswiss.md). In addition, I made a few minor comments and suggestions.

File: HISTORY.md (hunk 1 above)
Comment (nestabur) [reply HISTORY.md:13]: There is no package in PyPI repository, that entry in History is an old one ... shoulw we remove it?

File: src/pyflexplot/__init__.py
Comment (nestabur) [reply src/pyflexplot/__init__.py:1]: Should we add you as the main contact?

File: pyproject.toml [hunk 2]
```diff
@@ -24,23 +28,54 @@ classifiers = [
...
+[[tool.poetry.source]]
+name = "pypi-meteoswiss"
+url = "https://hub.meteoswiss.ch/nexus/repository/python-all/simple"
+priority = "primary"
+
+[[tool.poetry.source]]
+name = "pypi-mch-publish"
+url = "https://hub.meteoswiss.ch/nexus/repository/python-mch/"
+
+[tool.poetry.dependencies]
```
[the remark this answers is not in the document]
Comment (nestabur) [reply pyproject.toml:48]: Yes, since it is meant to be used with poetry in the development environment.

File: src/pyflexplot/__init__.py
Comment (victoria-cherkas) [reply src/pyflexplot/__init__.py:1]: There are about 9 __init__.py files in the repo, all with 
```
__author__ = "Stefan Ruedisuehli"
__email__ = "stefan.ruedisuehli@env.ethz.ch"
__version__ = "0.1.0"
```
but often the version is different per file.
I think these entries are unnecessary, also given that the authors/contacts are listed in pyproject.toml and AUTHORS.md.
@pirmink and @nestabur I suggest that I delete these blocks from the init files. Do you agree?

Comment (victoria-cherkas) [reply src/pyflexplot/__init__.py:1]: Added @pirmink as contact in pyproject.toml, and Stefan, @nestabur and I as contributors in AUTHORS.md: see https://github.com/MeteoSwiss-APN/pyflexplot/pull/41/commits/55619285bb69498c6522dfbde2ab2b684ecf864f

File: HISTORY.md (hunk 1 above)
Comment (victoria-cherkas) [reply HISTORY.md]: Edited HISTORY.md https://github.com/MeteoSwiss-APN/pyflexplot/pull/41/commits/f4dd6891e2f220ef55f101c03c0a924c02319148

**Comment by victoria-cherkas** [comment]
@pirmink I believe all the points above are addressed, please could you review once more and let us know if anything is still missing?

**Review by victoria-cherkas — APPROVED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -24,23 +28,52 @@ classifiers = [
...
-[project.urls]
-source = "https://github.com/MeteoSwiss-APN/pyflexplot"
-documentation = "https://github.io/MeteoSwiss-APN/pyflexplot"
-
-[project.scripts]
-# Format: <command> = "<package>.<module>:<function>"
-# pyflexplot = "pyflexplot.cli:main"
+[tool.poetry.scripts]
 pyflexplot  = "pyflexplot.cli.cli:cli"
 pytrajplot  = "pytrajplot.cli:cli"
```
Comment (pirmink) [inline pyproject.toml:34]: pytrajplot never became a part of this package, delete this line
```suggestion
```

File: pyproject.toml [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
-
-[project]
+[tool.poetry]
 name = "pyflexplot"
-version = "1.1.1"
+# Keep only Major and Minor version in this configuration,
+# the patch version is calculated when releasing the library
+version = "2.0"
 description = "PyFlexPlot - Visualize and post-process FLEXPART dispersion simulation results stored in NetCDF format"
+authors = ["Pirmin Kaufmann <pirmin.kaufmann@meteoswiss.ch>", "Stefan Ruedisuehli <stefan.ruedisuehli@env.ethz.ch>"]
```
Comment (pirmink) [inline pyproject.toml:7]: Stefan has written the vast majority of this code and should be the first author
```suggestion
authors = ["Stefan Ruedisuehli <stefan.ruedisuehli@env.ethz.ch>", "Pirmin Kaufmann <pirmin.kaufmann@meteoswiss.ch>"]
```

File: AUTHORS.md [hunk 5]
```diff
@@ -2,8 +2,10 @@
...
 
-- Stefan Ruedisuehli <stefan.ruedisuehli@env.ethz.ch>
+- Pirmin Kaufmann <Pirmin.Kaufmann@meteoswiss.ch>
 
 ## Contributors
 
-None yet. Why not be the first?
+- Stefan Ruedisuehli <Stefan.Ruedisuehli@env.ethz.ch>
+- Néstor Tarin Burriel <Nestor.TarinBurriel@meteoswiss.ch> 
+- Victoria Cherkas <Victoria.Cherkas@meteoswiss.ch>
```
Comment (pirmink) [inline AUTHORS.md:11]: Add Marco Ketzel who added the shapefile capability
```suggestion
- Victoria Cherkas <Victoria.Cherkas@meteoswiss.ch>
- Marco Ketzel
```

**Review by pirmink — COMMENTED** [review]
Few minor comments concerning authors and contributors.

**Comment by victoria-cherkas** [comment]
Thanks for the re-review @pirmink. Applied your suggestions 👍

**Review by pirmink — APPROVED** [review]

**Pull request merged**